src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java

```diff
@@ -19,156 +50,215 @@ void setup() {
+    when(response.getBody()).thenReturn(customTypesResult);
+
+    final ByProjectKeyGraphqlRequestBuilder byProjectKeyGraphqlRequestBuilder = mock();
+    when(sourceClient.graphql()).thenReturn(byProjectKeyGraphqlRequestBuilder);
+    final ByProjectKeyGraphqlPost byProjectKeyGraphqlPost = mock();
+    when(byProjectKeyGraphqlRequestBuilder.post(any(GraphQLRequest.class)))
+        .thenReturn(byProjectKeyGraphqlPost);
+    when(byProjectKeyGraphqlPost.execute()).thenReturn(CompletableFuture.completedFuture(response));
+
+    // test
+    final CompletionStage<List<CategoryDraft>> draftsFromPageStage =
+        categorySyncer.transform(categoryPage);
+
+    // assertions
+    final List<CategoryDraft> expectedResult =
+        toCategoryDrafts(sourceClient, referenceIdToKeyCache, categoryPage).join();
+    final List<String> referenceKeys =
+        expectedResult.stream()
+            .filter(category -> category.getCustom() != null)
+            .map(category -> category.getCustom().getType().getId())
+            .collect(Collectors.toList());
+    assertThat(referenceKeys).doesNotContainSequence(referenceIds);
+    assertThat(draftsFromPageStage).isCompletedWithValue(expectedResult);
+  }
+
+  @Test
+  void syncWithError_ShouldCallErrorCallback() {
+    // preparation: category with no key is synced
+    final ProjectApiRoot sourceClient = mock(ProjectApiRoot.class);
+    final ProjectApiRoot targetClient = mock(ProjectApiRoot.class);
+    final ByProjectKeyCategoriesRequestBuilder byProjectKeyCategoriesRequestBuilder = mock();
+    when(sourceClient.categories()).thenReturn(byProjectKeyCategoriesRequestBuilder);
+    final ByProjectKeyCategoriesGet byProjectKeyCategoriesGet = mock();
+    when(byProjectKeyCategoriesRequestBuilder.get()).thenReturn(byProjectKeyCategoriesGet);
+    when(byProjectKeyCategoriesGet.withSort(anyString())).thenReturn(byProjectKeyCategoriesGet);
+    when(byProjectKeyCategoriesGet.withLimit(anyInt())).thenReturn(byProjectKeyCategoriesGet);
+    when(byProjectKeyCategoriesGet.withWithTotal(anyBoolean()))
+        .thenReturn(byProjectKeyCategoriesGet);
+    final ApiHttpResponse<CategoryPagedQueryResponse> response = mock(ApiHttpResponse.class);
+    final List<Category> categories =
+        Collections.singletonList(readObjectFromResource("category-no-key.json", Category.class));
+    final CategoryPagedQueryResponse categoryPagedQueryResponse =
+        CategoryPagedQueryResponseBuilder.of()
+            .results(categories)
+            .limit(20L)
+            .offset(0L)
+            .count(1L)
+            .build();
+    when(response.getBody()).thenReturn(categoryPagedQueryResponse);
+    when(byProjectKeyCategoriesGet.execute())
+        .thenReturn(CompletableFuture.completedFuture(response));
+
+    // test
+    final CategorySyncer categorySyncer =
+        CategorySyncer.of(sourceClient, targetClient, mock(Clock.class));
```

Comment:
```suggestion
        CategorySyncer.of(sourceClient, mock(ProjectApiRoot.class), mock(Clock.class));
```
and remove targetClient variable.

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/498/commits/db7224267c69019001a9ea6c16f38fa021c87a5a